### ISSUE TITLE
Can't create gift via recordset when recordset too large

### ISSUE THREAD

**Issue by garrettgirard** [specify/specify7#127]
On KUFish - master branch

If you try to create a gift or a loan by using a recordset and the recordset has too many records, selecting that record won't open up the window that allows you to select which CO you want to gift/loan.

For reference, the recordset I tried to use had over 41,000 records.

**Comment by benanhalt** [comment]
Could you try a few different sizes 50, 100, 200, etc. and see how many it can do before things start taking to long?

**Comment by garrettgirard** [comment]
Number of records and how long they take: 10 ~ 1 second          50 ~ 2 seconds          100 ~ 4 seconds          250 ~ 10 seconds          500 ~ 20 seconds          1000 ~ 45 seconds          5000 ~ 9 minutes

So, it seems as if the recordset will work eventually, if you're willing to wait long enough.

**Comment by benanhalt** [comment]
Let's see how long it takes in Sp6.

**Comment by garrettgirard** [comment]
This is not much of a problem in Sp6. I created a recordset with over 31,000 preparations and it popped up almost immediately.

**Comment by benanhalt** [comment]
I looked into where are the performance bottlenecks and found a few things.
* Restructuring the `preps_available_rs` query gets that part down to just a couple seconds for ~40k records.
* The frontend then bogs down after receiving the results. Adding the jQuery-spinner to the thousands of fields takes longer than I cared to wait.
* Turning off the spinners allowed the dialog to finally be rendered in around a minute, but then it seemed like the business rules were bogging down when any values were changed.
